cpp/leiden_community_detection_module/leiden_utils/leiden_utils.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef LEIDEN_UTILS_HPP
+#define LEIDEN_UTILS_HPP
+
+#include <vector>
+#include <algorithm>
+#include <unordered_map>
+#include <omp.h>
+#include <utility>
+#include <set>
+#include <iterator>
+
+#include <boost/unordered_map.hpp>
+#include <boost/unordered_set.hpp>
+
+namespace leiden_alg {
+
+struct Graph {
+  std::vector<std::vector<std::uint64_t>> adjacency_list; // node_id -> neighbors
+
+  // Add an edge to the graph
+  inline void addEdge(std::uint64_t u, std::uint64_t v) {
+    if (u >= adjacency_list.size()) {
+        adjacency_list.resize(u + 1);
+    }
+    adjacency_list[u].push_back(v);
+  }
+
+  inline bool isVertexInGraph(std::uint64_t u) const {
+      return u < adjacency_list.size();
+    }
+
+  inline std::size_t size() const {
+      return adjacency_list.size();
+  }
+
+  inline const std::vector<std::uint64_t> &neighbors(std::uint64_t u) const {
+    return adjacency_list[u];
+  }
+};
```

Comment:
This looks very vanilla and something we already have in the global utils, or it should be there -> extract 🙏

Reply:
The structure has changed a bit. Even though we have a Graph in `mg_utils.hpp`, the graph there contains a few unnecessary parts and it's just easier to use this one at the moment. Perhaps we should just rename this one, or do you prefer the option of extracting and reusing the components?

Reply:
All good let's move with the specific implementation (what's already there), we'll merge in the future if needed.